### DISCUSSION TITLE
Localization of the e-Mail-text which are sent by several occasions

### DISCUSSION THREAD

**Discussion by theGitMichel** [dani-garcia/vaultwarden#3565] [Ideas]
Hi there, 

I wonder if there is a possibility to localize the text of the e-mails which are sent when e.g. a user logs into its account.
Currently, every mail is in English while the web vault is localized, in my case in Germany. 

Unfortunately, most of my users speak only one language (German), so the idea of translating these e-mail-texts raised. But I didn't find anything where to edit it. 

Maybe someone had the same idea and can give me a hint, where I can find all necessary?

**Comment by theGitMichel** [comment]
Sorry. After sending my idea, I found the right words for searching and finding the right discussion:
E-mail template language #3141: https://github.com/dani-garcia/vaultwarden/discussions/3141

**Reply by stefan0xC** [reply]
Also check out the Wiki page: [Translating the email templates](https://github.com/dani-garcia/vaultwarden/wiki/Translating-the-email-templates). @kennymc-c already [translated the E-Mail templates into german](https://github.com/kennymc-c/vaultwarden-lang-de).

**Reply by theGitMichel** [reply]
I think you've lost me now. 
I installed vaultwarden by a docker-compose.yml as docker container. But, i can't find any folders "src/static/templates/email". 
Do you have a hint how to get it?

**Reply by BlackDex** [reply]
https://github.com/dani-garcia/vaultwarden/tree/adf67a8ee887d21e104adfa8b6521d7971f5a1f1/src/static/templates/email

**Reply by theGitMichel** [reply]
Yes. That's clear. @BlackDex :)
I've meant I didn't find this path on my server, where I'e deployed the docker container of vaultwarden.
I am still learning these things like docker-compose and how to combine different images etc. 
So I got the solution presented, but my synapses currently can't do anything with it. That's why I asked for a hint. :)

**Reply by BlackDex** [reply]
Check: https://github.com/dani-garcia/vaultwarden/blob/adf67a8ee887d21e104adfa8b6521d7971f5a1f1/.env.template#L49-L53

It needs at least the `templates` path as main path, so that it can see the `email` and `admin` folders, and the `404.hbs`.
You can mount that via an `-v` option with docker, you also do not need all the files, only the mail folder is fine, it will take the rest from the built-in files. You also do not need to have all the email template files, just using the files you want to adjust is fine.

**Reply by theGitMichel** [reply]
okay, I've tried this:
- cloned the repo from here: https://github.com/kennymc-c/vaultwarden-lang-de.git
- renamed the folder to data/templates/email
- edited docker-compose.yml: add row under "environment:" > - TEMPLATES_FOLDER=./data/templates/email
- restart via docker compose down && docker compose up -d

But nothing happens. Mails are in english.

**Reply by BlackDex** [reply]
If you use docker, and your data folder is `/data/` is suggest to remove the `.` in front of `./data/templates/email`
Also, as mentioned,  you need to use the `templates` folders as main, because it needs to see the folders `email` and/or `admin`.
So, it should be `TEMPLATES_FOLDER=/data/templates` as variable.

**Reply by theGitMichel** [reply]
No, unfortunately it doesn't show the edited version. 
I only get the embedded one in English, with no changes.

**Reply by theGitMichel** [reply]
yes i still tried it before without the . and even without the / and with and without the "email". 
I checked if I had written something wrong, but all is right. 

the path of the e-mail-folder is ../vaultwarden/data/templates/email

so, for dummies, a really stupid question:
the path in docker-compose.yml should be:
- TEMPLATES_FOLDER=data/templates
- TEMPLATES_FOLDER=/data/templates

**Comment by theGitMichel** [comment]
I tried also this:
- cloned the complete repo of vaultwarden
- copied the folders /templates/.. to /vaultwarden/vw-data/templates
- I changed the two files of smtp_test
- in docker-compose.yml I added in environment-section the row - TEMPLATES_FOLDER=vw-data/templates/email
- restart via docker compose down && docker compose up -d
- send a test mail from the admin panel 
- but nothing happens, the "hello world" i've added wasn't visible in the mail. 

maybe i should give up...

**Reply by BlackDex** [reply]
The env variables needs to be from the point of view from within the container, not from the host.

So, if you use `-v /vaultwarden/vw-data:/data` as your volume mount, the the env needs to be pointing to `/data/templates`

**Comment by CamelT0E** [comment]
I just unzip, rename to "template" and copy folder in the root directory of mounted vaultwarden folder. This works without changing the path on my synology diskstation

![grafik](https://github.com/dani-garcia/vaultwarden/assets/59985723/f8d7f3d6-facf-45ee-b753-ce61f513338c)

![grafik](https://github.com/dani-garcia/vaultwarden/assets/59985723/99deddf1-458d-43b8-a129-cec551a102c2)

**Reply by theGitMichel** [reply]
Das ist eine gute Idee. Ich Probier es mal auf der DiskStation und bau es dann wenn es funktioniert auf dem Root Server nach.

**Comment by theGitMichel** [comment]
What the hell, what a torture! I finally got it! :)

The method with environment: TEMPLATES_FOLDER=/vw-data/templates doesn't work for me. 

But by binding a volume like this it works in the end:
volumes:
- 'opt/containers/vaultwarden/vw-data/templates/:/data/templates'

Thank you very much for your help and patience!

**Reply by CamelT0E** [reply]
Nice that it works for you.
But why do you need to mount the templates folder extra? You should have data already, shouldn't you? So you just have to copy the templates folder to vw-data. An extra mount is not necessary.

**Reply by theGitMichel** [reply]
You're totally right! 
Just tried it again without mounting or the additional environment row "TEMPLATES_FOLDER" and ... bang! It works, too!

Manchmal stell ich mich einfach viel zu dämlich an. 🥇
